compiler/rustc_typeck/src/check/callee.rs

```diff
@@ -374,7 +374,24 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
                                     |p| format!("`{}` defined here returns `{}`", p, callee_ty),
                                 )
                             }
-                            _ => Some(format!("`{}` defined here", callee_ty)),
+                            _ => {
+                                match def {
+                                    // Emit a different diagnostic for local variables, as they are not
+                                    // type definitions themselves, but rather variables *of* that type.
+                                    Res::Local(hir_id) => Some(format!(
+                                        "`{}` has type `{}`",
+                                        self.tcx.hir().name(hir_id),
+                                        callee_ty
+                                    )),
+                                    Res::Def(kind, def_id) if let Some(Namespace::ValueNS) = kind.ns() => {
```

Comment:
I think this can just be changed to:

```suggestion
                                    Res::Def(kind, def_id) if kind.ns() == Some(Namespace::ValueNS) => {
```

(with the feature gate and `allow(incomplete_features)` removed as well)